NCMP_for_2-1-1_and_2-2-1/WHO_code_test.R

```diff
@@ -0,0 +1,97 @@
+NCMP_data <- read.csv("../ncmp_1819_final_non_disclosive_published.csv")
+
+
+# truncating the age in months variable, so 49.9 is classed as 49
+# (that is, similarly to rounding age, you are not 30 until 29th year has passed)
+NCMP_data$AgeInMonths <- trunc(NCMP_data$AgeInMonths) 
+
+# retaining only age groups of interest (under 5s, exclusing 5)
+NCMP_data <- NCMP_data[NCMP_data$AgeInMonths < 60, ]
+if((max(NCMP_data$AgeInMonths) > 59) | (min(NCMP_data$AgeInMonths) < 48)){
+  print("ERROR! Check AgeInMonths is in expected range (between 48 and 59 months)")}
+
+# clean up suppressions from dataset (not necessary when working with full data set)
+clean_NCMP <- NCMP_data[NCMP_data$Height > 0, ]
+
+# remove rows with suppressed imd and perform check that imd is in expected range
+clean_NCMP <- clean_NCMP[clean_NCMP$suppress_imd != "Y", ]
+if (any(unique(clean_NCMP$schoolindexofmultipledepriv) %in% c(1:10)) == FALSE){
+    print("ERROR! Check that schoolindexofmultipledepriv variable ranges from 1 to 10")
+  }
+
+#check how many entries were lost after all the cleaning:
+# This is just for testing purposes, raw data should need no cleaning
+lost_data <- paste("Cases:", nrow(clean_NCMP)-nrow(NCMP_data), ",", 
+                   "Percent:", (nrow(NCMP_data)-nrow(clean_NCMP)) / nrow(NCMP_data) * 100) # as above, in %
+lost_data
+
+# regroup imd variable from deciles into quintiles to fit the WHO function requirement
+clean_NCMP$school_imd_quintile <- ceiling(clean_NCMP$schoolindexofmultipledepriv / 2)
+
+if (any(unique(clean_NCMP$school_imd_quintile) %in% c(1:5)) == FALSE){
+  print("ERROR! Check that school_imd_quintile variable ranges from 1 to 5")
+}
+
+
+#main function, takes around 26 mins to run:
+
+#the wealthq argument is fed the regrouped IMD variable and assumes 1 = most deprived, 5 = least deprived
+#typeres argument expects urban/rural breakdown variable, 
+# which is not available in the test data but is reported to be available in the raw data as "PupilUrbanRuralIndicator"
+# schoolgovernmentofficeregion variable is used here as a test for othergr argument (which may take ethnicity from the raw data)
+# reformatting this variable into character, in case it's a factor (to avoid function errors)
+clean_NCMP$schoolgovernmentofficeregion <- as.character(clean_NCMP$schoolgovernmentofficeregion)
+
+test_NCMP <- anthro_prevalence(sex = clean_NCMP$GenderCode,
+                             age = clean_NCMP$AgeInMonths, 
+                             is_age_in_month = TRUE, 
+                             weight = clean_NCMP$Weight, 
+                             lenhei = clean_NCMP$Height, 
+                             measure = "H",
+                             wealthq = clean_NCMP$school_imd_quintile,
+                             othergr = clean_NCMP$schoolgovernmentofficeregion)
+
+#get relevant info only
+rownames(test_NCMP) <- test_NCMP$Group #assigning the first column as rownames (makes is cleaner)
```

Comment:
Pretty sure your approach creates a better format output for them to publish. I just wrote it so that it gives up our csv format but we use that code on what they publish :)